### DISCUSSION TITLE
Android - OpenThread Border Router on VIM3 Khadas SBC

### DISCUSSION THREAD

**Discussion by ravichoudhary321** [openthread/openthread#9948] [Q&A] [answered]
Hi everybody,

I need to setup the OTBR in vim3 Khadas SBC.

 I have cloned the ot-br-posix code from Android codespace.
**https://cs.android.com/android/platform/superproject/main/+/main:external/ot-br-posix/_**

Now, i need to compile/build the cloned OTBR posix android code and need to flash in VIM3 Khadas.

I would appreciate it if anyone could provide me with information on how Android OTBR posix code can flash in VIM3 Khadas...!

**Comment by wgtdkp** [comment, marked as answer]
You shouldn't try to build ot-br-posix separately for Android.  Instead, you should clone the  entire AOSP codebase and build a new image from source. See https://android-review.git.corp.google.com/c/platform/packages/modules/Connectivity/+/3007440 for a guide to try with cuttlefish device instead.

Additionally, to support vim3 Khadas SBC, you (or ask the vendor) need to create an implementation for the Android Thread HAL API: https://cs.android.com/android/platform/superproject/main/+/main:hardware/interfaces/threadnetwork/aidl/aidl_api/. An example implementation can be found in https://cs.android.com/android/platform/superproject/main/+/main:hardware/interfaces/threadnetwork/aidl/default/

**Reply by lmnotran** [reply]
Hi @wgtdkp, Is there any way for you to export the contents of https://android-review.git.corp.google.com/c/platform/packages/modules/Connectivity/+/3007440? I'm unable to access it as the link is behind Google's Corp SSO

**Reply by jwhui** [reply]
Here's the guide:

## Try Thread with Cuttlefish

```
# Get the code and go to the Android source code root directory

source build/envsetup.sh
lunch aosp_cf_x86_64_phone-trunk_staging-userdebug
m

launch_cvd

Open `https://localhost:8443/` in your web browser, you can find the Thread
demoapp (with the Thread logo) in the cuttlefish instance. Open it and have fun with Thread!

**Reply by lmnotran** [reply]
Thanks @jwhui !
